Client/Networking/NetworkSession.swift

```diff
@@ -3,10 +3,17 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import Foundation
+import Combine
 
 typealias NetworkSessionDataResponse = (Data, URLResponse)
 
 protocol NetworkSession {
+  func dataRequest(with url: URL, _ completion: @escaping (Result<NetworkSessionDataResponse, Error>) -> Void)
+  func dataRequest(with urlRequest: URLRequest, _ completion: @escaping (Result<NetworkSessionDataResponse, Error>) -> Void)
+  
+  func dataRequest(with url: URL) -> AnyPublisher<NetworkSessionDataResponse, Error>
+  func dataRequest(with urlRequest: URLRequest) -> AnyPublisher<NetworkSessionDataResponse, Error>
+  
   func dataRequest(with url: URL) async throws -> NetworkSessionDataResponse
   func dataRequest(with urlRequest: URLRequest) async throws -> NetworkSessionDataResponse
```

Comment:
I want to keep them for now because I want to run some tests some time to see what is causing the issue and I'm hoping with an upgraded Xcode it will either crash or give an error or something.